tests/integration/test_issue_2355.py

```diff
@@ -73,28 +67,47 @@ def test_ssl_context(
 
     work_dir = os.path.join(str(tmpdir), "work_dir")
     os.mkdir(work_dir)
+    docker_run_args = [
+        "docker",
+        "run",
+        "--rm",
+        "-v",
+        "{pex_project_dir}:/code".format(pex_project_dir=pex_project_dir),
+        "-v",
+        "{work_dir}:/work".format(work_dir=work_dir),
+        "-w",
+        "/code",
+        "test_issue_2355",
+        "/python/bin/python3.9",
+        "-mpex.cli",
+    ]
+    complete_platform_data = json.loads(
+        subprocess.check_output(
+            args=docker_run_args + ["interpreter", "inspect", "--markers", "--tags"]
+        )
+    )
+    container_interpreter = CompletePlatform.create(
+        marker_environment=MarkerEnvironment(**complete_platform_data["marker_environment"]),
+        supported_tags=CompatibilityTags.from_strings(complete_platform_data["compatible_tags"]),
+    )
+    pip_version = try_(
+        compatible_version(
+            targets=Targets(
+                interpreters=tuple([PythonInterpreter.get()]),
+                complete_platforms=tuple([container_interpreter]),
+            ),
+            requested_version=PipVersion.DEFAULT,
+            context=__name__ + ".test_ssl_context",
+        )
+    )
```

Comment:
N.B.: We need the lock created in the container to use a Pip version compatible with the tests interpreter. Previously, the one case where the default did not satisfy this constraint was handled by plumbing `--env _PEX_PIP_VERSION`. Now, we calculate a compatible version up front instead.